### ISSUE TITLE
[Bug]: Exported PDF figures cannot be modified by Adobe Illustrator correctly.

### ISSUE THREAD

**Issue by JerrySoybean** [matplotlib/matplotlib#27893]
### Bug summary

Use `imshow` to generate a figure, and then save it as a `pdf` file. If you open the generated pdf figure in Adobe Illustrator, the figure in Adobe Illustrator becomes different. Colors are wrongly assigned.

### Code for reproduction

```Python
import numpy as np
import matplotlib.pyplot as plt

import matplotlib
print(matplotlib.__version__)

a = np.array([[1, 2, 3], [4, 5, 6]])
```

### Actual outcome

![image](https://github.com/matplotlib/matplotlib/assets/44856732/5d61986b-2c83-4f01-a117-4a13a4b22dda)

### Expected outcome

![image](https://github.com/matplotlib/matplotlib/assets/44856732/4e15888e-7b49-49d8-b1b3-4eddadc8e666)

### Matplotlib Version

3.8.3

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.10.13

### Jupyter version

7.0.8

### Installation

conda

**Comment by jklymak** [comment]
Please show your full code.  Also do other viewers work?

**Comment by JerrySoybean** [comment]
> Please show your full code. Also do other viewers work?

That's all the code I use. Only Adobe Illustrator doesn't work correctly.

**Comment by timhoffm** [comment]
> > Please show your full code. Also do other viewers work?
> 
> That's all the code I use. Only Adobe Illustrator doesn't work correctly.

The shown code is certainly not complete. It only imports matplotlib and creates a numpy array. It does not call any matplotlib functions.

**Comment by JerrySoybean** [comment]
> > > Please show your full code. Also do other viewers work?
> > 
> > 
> > That's all the code I use. Only Adobe Illustrator doesn't work correctly.
> 
> The shown code is certainly not complete. It only imports matplotlib and creates a numpy array. It does not call any matplotlib functions.

Sorry, I’ve updated the code. Thanks for this reminding.

**Comment by jklymak** [comment]
I can reproduce this with Illustrator 2023.  But given that every other viewer handles this properly, including Acrobat and Photoshop, my tendency is to blame Illustrator.

**Comment by tacaswell** [comment]
Can someone with a valid illustrator license report this as a bug to them?  The pdf spec is complicated enough that it is possible that they have a bug, but it is also possible that we are emitting technically incorrect files and the other viewers are all buggy-in-a-useful-way or there is some undefined behavior gets lucky most of the time.

My guess is that something is going wrong with images use color palates which we recently fixed some bug with.  Does this work with older versions of mpl?

**Comment by jklymak** [comment]
Fair enough - this bisects to PR #25704. Before 8d42212418 the colors look fine in Illustrator

**Comment by JerrySoybean** [comment]
> I can reproduce this with Illustrator 2023. But given that every other viewer handles this properly, including Acrobat and Photoshop, my tendency is to blame Illustrator.

But the thing is, it's because of the upgrade of matplotlib that changes the outcome in Illustrator? And I'm not sure can acrobat and Photoshop modify PDF as the way used in Illustrator?

**Comment by QuLogic** [comment]
> Fair enough - this bisects to PR #25704. Before [8d42212](https://github.com/matplotlib/matplotlib/commit/8d4221241851795aa7cd9bdf083eb9f65f9bcaf1) the colors look fine in Illustrator

Hmm, are you sure this is the correct commit? If I run the original code, I get the same PDF output (other than the creator metadata).

**Comment by jklymak** [comment]
Not 100% sure.  Bisecting a problem like this is a lot of manual opening and closing of files so I may have messed up

**Comment by mxposed** [comment]
I have run into this bug. Attaching 2 pdfs generated with the same code, from v 3.7.2 (which I confirm the issue in Illustrator 2024) and from 3.3.4 where there is no issue

[test_3.7.2.pdf](https://github.com/user-attachments/files/21008739/test_3.7.2.pdf)
[test_3.3.4.pdf](https://github.com/user-attachments/files/21008740/test_3.3.4.pdf)
